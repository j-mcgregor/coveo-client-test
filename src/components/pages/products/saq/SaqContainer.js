import React from 'react';
import { Route, Switch, useRouteMatch } from 'react-router-dom';
import ProductsIndex from './views/index/IndexConnect';
import ProductsShow from './views/show/ShowConnect';

const ProductsContainer = () => {
  const { path } = useRouteMatch();

  return (
    <div>
      <Switch>
        <Route exact path={path}>
          <ProductsIndex />
        </Route>
        <Route path={`${path}/:id`}>
          <ProductsShow />
        </Route>
      </Switch>
    </div>
  );
};

export default ProductsContainer;
