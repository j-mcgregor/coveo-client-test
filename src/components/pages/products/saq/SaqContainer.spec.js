/* eslint-disable react/prop-types */
import React from 'react';
import { shallow } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import SaqContainer from './SaqContainer';

jest.mock('react-router', () => ({
  useRouteMatch: jest.fn().mockReturnValue({ path: '/catalog' }),
  withRouter: jest.fn()
}));

describe('SaqContainer', () => {
  it('should render', () => {
    const wrapper = shallow(
      <ThemeProvider theme={{ light: { color: 'yellow' } }}>
        <SaqContainer />
      </ThemeProvider>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
