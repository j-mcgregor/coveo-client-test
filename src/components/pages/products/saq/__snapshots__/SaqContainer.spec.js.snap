// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SaqContainer should render 1`] = `ShallowWrapper {}`;
